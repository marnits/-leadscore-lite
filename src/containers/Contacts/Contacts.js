import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import List from '../../components/List';
import Text from '../../components/Text';
import ListItem from '../../components/List/components/ListItem';
import styles from './styles.module.scss';
import contactType from '../../utils/propTypes/contact';

const columns = [{
  name: 'Name',
  size: 3,
  align: 'left',
}, {
  name: 'Email',
  size: 2,
  align: 'left',
}, {
  name: 'Phone Number',
  size: 1,
  align: 'right',
}];

class Contacts extends PureComponent {
  componentDidMount() {
    const { fetchContacts } = this.props;

    fetchContacts();
  }

  render() {
    const {
      contactsById,
      contactsByHash,
      fetchContacts,
      loading,
      last,
    } = this.props;

    return (
      <div className={styles.list}>
        <List
          title="Contacts"
          columns={columns}
          loadMore={fetchContacts}
          loading={loading}
          last={last}
        >
          {contactsById.map(id => (
            <ListItem key={id}>
              <Text strong text={contactsByHash[id].name} />
              <Text text={contactsByHash[id].email} />
              <Text text={contactsByHash[id].phoneNumber} />
            </ListItem>
          ))}
        </List>
      </div>
    );
  }
}

Contacts.propTypes = {
  last: PropTypes.bool.isRequired,
  loading: PropTypes.bool.isRequired,
  fetchContacts: PropTypes.func.isRequired,
  contactsById: PropTypes.arrayOf(PropTypes.string).isRequired,
  contactsByHash: PropTypes.objectOf(contactType).isRequired,
};

export default Contacts;
