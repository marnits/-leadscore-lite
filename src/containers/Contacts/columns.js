export default [{
  name: 'Name',
  size: 3,
  align: 'left',
}, {
  name: 'Email',
  size: 2,
  align: 'left',
}, {
  name: 'Phone Number',
  size: 1,
  align: 'right',
}];
