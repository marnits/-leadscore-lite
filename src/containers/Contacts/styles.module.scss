.wrapper {
  background-color: #f9f8f8;
  height: 100vh;
  width: 100%;
  flex-wrap: nowrap;
  display: flex;
  flex-basis: auto;
}

.navbar {
  height: 30px;
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  padding: 8px;
}
