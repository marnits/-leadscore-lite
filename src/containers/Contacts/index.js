import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Contacts from './Contacts';
import { fetchContactsRequest } from '../../actions/contacts';

const mapStateToProps = ({
  contacts: {
    byId,
    byHash,
    loading,
    total,
    offset,
  },
}) => ({
  contactsById: byId,
  contactsByHash: byHash,
  loading,
  last: total === offset,
});

const mapDispatchToProps = dispatch => ({
  fetchContacts: bindActionCreators(fetchContactsRequest, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(Contacts);
