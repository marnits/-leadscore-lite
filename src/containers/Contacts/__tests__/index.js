import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Contacts from '../Contacts';

const props = {
  contactsById: ['123'],
  contactsByHash: {
    123: {
      id: '123',
      name: 'Test Test',
      email: 'abc@xyz.io',
      phoneNumber: '+48 423 231 543',
    },
  },
  last: false,
  loading: false,
  fetchContacts: () => {},
};

describe('Contacts', () => {
  test('should call fetchContacts on componentDidMount', () => {
    const spy = jest.spyOn(props, 'fetchContacts');
    const wrapper = shallow(<Contacts {...props} />);

    wrapper.instance().componentDidMount();
    expect(spy).toHaveBeenCalled();
  });

  test('should properly render all children', () => {
    const tree = renderer.create(<Contacts {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
