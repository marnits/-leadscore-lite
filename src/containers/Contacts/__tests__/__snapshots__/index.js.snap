// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts should properly render all children 1`] = `
<div
  className="list"
>
  <div
    className="wrapper"
  >
    <div
      className="title"
    >
      <span
        className="Header"
      >
        Contacts
      </span>
    </div>
    <div
      className="row"
    >
      <div
        className="
                column-name
                align-content-left
                flex-3
              "
      >
        <label
          className="Label"
          htmlFor={null}
        >
          Name
        </label>
      </div>
      <div
        className="
                column-name
                align-content-left
                flex-2
              "
      >
        <label
          className="Label"
          htmlFor={null}
        >
          Email
        </label>
      </div>
      <div
        className="
                column-name
                align-content-right
                flex-1
              "
      >
        <label
          className="Label"
          htmlFor={null}
        >
          Phone Number
        </label>
      </div>
    </div>
    <div
      className="list"
      onScroll={[Function]}
    >
      <div
        className="row"
      >
        <div
          className="
          column
          align-content-left
          flex-3
        "
        >
          <span
            className="text strong"
          >
            Test Test
          </span>
        </div>
        <div
          className="
          column
          align-content-left
          flex-2
        "
        >
          <span
            className="text false"
          >
            abc@xyz.io
          </span>
        </div>
        <div
          className="
          column
          align-content-right
          flex-1
        "
        >
          <span
            className="text false"
          >
            +48 423 231 543
          </span>
        </div>
      </div>
    </div>
    <div
      className="shadow"
      style={
        Object {
          "width": 0,
        }
      }
    />
  </div>
</div>
`;
